osconfig_tests/test_suites/package_management/package_management_test_data.go

```diff
@@ -100,49 +102,58 @@ func addPackageInstallTest(pkgTestSetup []*packageManagementTestSetup, testProje
+				ss := getPackageInstallStartupScript(tuple.pkgManager, packageName)
+				setup := NewPackageManagementTestSetup(image, instanceName, testName, vs, oc, assign, ss, vf)
+				pkgTestSetup = append(pkgTestSetup, setup)
+			}
 		case "rhel":
-			image = rhelImage
-			pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
-			oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, osconfigserver.BuildYumPackageConfig(pkgs, nil, nil), nil, nil, nil)
-			vs = fmt.Sprintf(packageInstalledString)
+			for _, image := range rhelImages {
+				pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
+				oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, osconfigserver.BuildYumPackageConfig(pkgs, nil, nil), nil, nil, nil)
+				vs = fmt.Sprintf(packageInstalledString)
+				instanceName := fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix)
+				assign := osconfigserver.BuildAssignment(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, osconfigserver.BuildInstanceFilterExpression(instanceName), []string{fmt.Sprintf("projects/%s/osConfigs/%s", testProjectConfig.TestProjectID, oc.Name)})
+				ss := getPackageInstallStartupScript(tuple.pkgManager, packageName)
+				setup := NewPackageManagementTestSetup(image, instanceName, testName, vs, oc, assign, ss, vf)
+				pkgTestSetup = append(pkgTestSetup, setup)
+			}
 		case "windows":
-			image = windowsImage
-			pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
-			oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, nil, osconfigserver.BuildGooPackageConfig(pkgs, nil, nil), nil, nil)
-			vs = fmt.Sprintf(packageInstalledString)
+			for _, image := range windowsImages {
+				pkgs := []*osconfigpb.Package{osconfigserver.BuildPackage(packageName)}
+				oc = osconfigserver.BuildOsConfig(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, nil, nil, osconfigserver.BuildGooPackageConfig(pkgs, nil, nil), nil, nil)
+				vs = fmt.Sprintf(packageInstalledString)
+				instanceName := fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix)
+				assign := osconfigserver.BuildAssignment(fmt.Sprintf("%s-%s-%s", path.Base(image), testName, uniqueSuffix), desc, osconfigserver.BuildInstanceFilterExpression(instanceName), []string{fmt.Sprintf("projects/%s/osConfigs/%s", testProjectConfig.TestProjectID, oc.Name)})
+				ss := getPackageInstallStartupScript(tuple.pkgManager, packageName)
+				setup := NewPackageManagementTestSetup(image, instanceName, testName, vs, oc, assign, ss, vf)
+				pkgTestSetup = append(pkgTestSetup, setup)
```

Comment:
Here too, I really like the approach, but there's a bunch of duplicate boiler plate code that could be refactored.

Reply:
i really wanted to refactor out the common code, but the only difficulty is BuildOsConfig utility method. This method is tightly coupled with the type of test case and the package manager of the platform.
To refactor out the BuildOsConfig call, i either might have to maintain a static mapping of package-manager and testcase to the function or i am thinking i can use builder pattern to build the osconfig.
Since i haven't really thought through this fully, how about i add this as a TODO task for refactoring, for now?